iib/workers/tasks/build_containerized_add.py

```diff
@@ -0,0 +1,376 @@
+        artifact_index_db_file = fetch_and_verify_index_db_artifact(
+            from_index=str(from_index),
+            temp_dir=temp_dir,
+        )
+
+        if from_index:
+            msg = 'Checking if bundles are already present in index image'
+            log.info(msg)
+            set_request_state(request_id, 'in_progress', msg)
+
+            with set_registry_token(overwrite_from_index_token, from_index_resolved, append=True):
+                present_bundles, present_bundles_pull_spec = _get_present_bundles(
+                    localized_git_catalog_path, temp_dir
+                )
+
+            filtered_bundles = _get_missing_bundles(present_bundles, resolved_bundles)
+            excluded_bundles = [
+                bundle for bundle in resolved_bundles if bundle not in filtered_bundles
+            ]
+            resolved_bundles = filtered_bundles
+
+            if excluded_bundles:
+                log.info(
+                    'Following bundles are already present in the index image: %s',
+                    ' '.join(excluded_bundles),
+                )
+
+        # This is a replacement for opm_registry_add_fbc for a containerized version of IIB.
+        # Note: only index.db is modified (FBC directory is unchanged)
+        _opm_registry_add(
+            base_dir=temp_dir,
+            index_db=artifact_index_db_file,
+            bundles=resolved_bundles,
+            overwrite_csv=(prebuild_info['distribution_scope'] in ['dev', 'stage']),
+            graph_update_mode=graph_update_mode,
+        )
+
+        deprecation_bundles = get_bundles_from_deprecation_list(
+            present_bundles_pull_spec + resolved_bundles, deprecation_list or []
+        )
+
+        arches = prebuild_info['arches']
+        if deprecation_bundles:
+            deprecate_bundles_fbc_containerized(
+                bundles=deprecation_bundles,
+                base_dir=temp_dir,
+                binary_image=prebuild_info['binary_image'],
+                index_db_file=artifact_index_db_file,
+            )
+
+        os.makedirs(os.path.join(temp_dir, 'from_db'), exist_ok=True)
+        # get catalog from SQLite index.db (hidden db) - not opted in operators
+        catalog_from_db, _ = opm_migrate(
+            index_db=artifact_index_db_file,
+            base_dir=os.path.join(temp_dir, 'from_db'),
+            generate_cache=False,
+        )
+
+        catalog_from_index = localized_git_catalog_path
+
+        # we have to remove all `deprecation_bundles` from `catalog_from_index`
+        # before merging catalogs otherwise if catalog was deprecated and
+        # removed from `index.db` it stays on FBC (from_index)
+        # Therefore we have to remove the directory before merging
+        for deprecate_bundle_pull_spec in deprecation_bundles:
+            # remove deprecated operators from FBC stored in index image
+            deprecate_bundle = get_image_label(
```

Comment:
Consider renaming `deprecate_bundle` to `deprecate_bundle_package`

Consider using Path. It's seems to be a best practice. We should also change it in other PRs.
```
if deperecate_bundle_package in Path(path_string).parts:
    shutil.rmtree(bundle_from_index)
```

Reply:
Changed